Made a tag to help people fix an empty file JSON error. (#1393)
python-discord/bot: bot/resources/tags/empty-json.md

When using JSON, you might run into the following error:
```
JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
This error could have appeared because you just created the JSON file and there is nothing in it at the moment.

> Whilst having the data empty is no problem, the file itself may never be completely empty.

Reviewer: Maybe it's just me, but I found the original phrasing a bit awkward.
```suggestion
Whilst having empty data is no problem, the file itself may never be completely empty.
```

> You most likely wanted to structure your JSON as a dictionary. To do this, change your JSON to read `{}`.

Reviewer: It can be more explicit about what it's trying to convey
```suggestion
You most likely wanted to structure your JSON as a dictionary. To do this, edit your empty JSON file so that it instead contains `{}`.
```

> Different data types are also supported. If you wish to read more on these, please refer to the following article: https://www.tutorialspoint.com/json/json_data_types.htm

Reviewer 2: ```suggestion
Different data types are also supported. If you wish to read more on these, please refer to [this article](https://www.tutorialspoint.com/json/json_data_types.htm).
```
Making this a text link hides the long link, saving some space

> Different data types are also supported. If you wish to read more on these, please reffer to the following article: https://www.tutorialspoint.com/json/json_data_types.htm

Reviewer 3: ```suggestion
Different data types are also supported. If you wish to read more on these, please refer to the following article: https://www.tutorialspoint.com/json/json_data_types.htm
```

When creating a new JSON file you might run into the following error.

`JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

> In short this error means your JSON is invalid in it's current state.

Reviewer 4: While true, this statement may be unnecessarily technical for some subset of the intended audience, who already know that their JSON isn't working. The line that follows it will probably be enough.

Author: Outdated. Has more info on the description below it now.

A JSON may never be completely empty and must always at least have one of the following items.

```

> object

Reviewer 4: An object is the dict-like structure that you can have in a JSON, yes? I think it would be easier for most of the intended readers to describe the valid structures in terms of their Python analogues, since the goal isn't to teach anyone anything about JSONs beyond how to debug this error.

Author: In the current text below it, says it's common to use {} to make an object, which is similar to a dictionary in python. Should I move this line up?

Reviewer 5: Even if it is the correct term, I am also of the mindset that we use python keywords. The tags in my opinion are by no means an extensive resource, but more of a quick overview to explain a concept, and set someone on the right path of looking into it.

The source you included uses `object`. If someone is interested in learning more about JSON, they can follow that and learn more. It does not make sense to have it that far removed from python though when the original question is probably in python.

Author: > Even if it is the correct term, I am also of the mindset that we use python keywords. The tags in my opinion are by no means an extensive resource, but more of a quick overview to explain a concept, and set someone on the right path of looking into it.
> 
> The source you included uses `object`. If someone is interested in learning more about JSON, they can follow that and learn more. It does not make sense to have it that far removed from python though when the original question is probably in python.

Ah right now I understand what you are saying. In the discord chat I didn't completely understand what you were saying. Now with context I understand more. I will fix this.

array
string
number

> "true"

Reviewer 4: Aren't `true`, `false`, and `null` valid literals in a JSON? If so, why are they in quotes here? It might be better to have an example.

Author: You might be looking at an old commit. In the latest one they are removed. It's purely list and object now.

"false"
"null"
```

To resolve this issue, you create one of the above values in your JSON. It is very common to use `{}` to make an object. Adding the following to your JSON should resolve this issue.

```json
{

> }

Reviewer 4: Does this mean that the JSON in this code example, if saved exactly as-is in a .json file, would be valid because it has the extra line? That's my takeaway but that seems to conflict with what else is conveyed in this message.

Author: That extra white line between the brackets is removed.
As stated earlier when it's COMPLETELY empty (like when it's just created) it's invalid. But this makes it have at least SOME content to make it valid.
Mayb there is a better way to word it? or maybe an extra note to add?
